plugins/uefi/fu-plugin-uefi.c

```diff
@@ -592,7 +592,13 @@ fu_plugin_startup (FuPlugin *plugin, GError **error)
 {
 	const guint8 *data;
 	gsize sz;
-	g_autoptr(GBytes) bios_information = fu_plugin_get_smbios_data (plugin, 0);
+	g_autoptr(GBytes) bios_information = NULL;
+
+	/* Silicom + Minnowboard Turbot D0/D1 PLATFORM */
+	if (fu_plugin_check_hwid (plugin, "386c2f52-44ec-55d3-b802-47631bfb8451"))
+		return TRUE;
```

Comment:
Not possible to do from quirk instead?
Also, can you please file something to Minnowboard folks so that they can fix it in a future FW upgrade and note that here?

Reply:
> Not possible to do from quirk instead?

Well, it's per-plugin, not per-device -- and needed before coldplug runs. I guess we could do something like superio, where we feed in all the hwids to `fu_plugin_lookup_quirk_by_id` with a key like `ForceEnable` or `AlwaysStart` (better names welcome). A quirk is probably the right way to do this as then we can easily add support for other machines "in the field" by dropping in an extra file.

> Also, can you please file something to Minnowboard folks

I've already sent email, I'm hoping new firmware will fix this. When there's an accepted issue I'll update this PR.

Reply:
It's a little hacky, but another thing you can do is if that check failed to see if `/sys/firmware/uefi` existed to get around this issue but issue a warning that there is a FW bug.

Reply:
(And of course if it doesn't exist the same normal failure)

Reply:
Something like the fixup? 100% untested.

Reply:
> another thing you can do is if that check failed to see if /sys/firmware/uefi existed

Done in the 2nd fixup. Also untested.